src/main/scala/mesosphere/marathon/state/TaskFailureRepository.scala

```diff
@@ -1,17 +1,20 @@
 package mesosphere.marathon.state
 
-import scala.collection.mutable
+import mesosphere.marathon.metrics.Metrics
 
-class TaskFailureRepository(store: EntityStore[TaskFailure], maxVersions: Option[Int] = Some(1)) {
+import scala.concurrent.Future
 
-  protected[this] val taskFailures = mutable.Map[PathId, TaskFailure]()
+class TaskFailureRepository(
+  protected val store: EntityStore[TaskFailure],
+  protected val maxVersions: Option[Int] = Some(1),
+  protected val metrics: Metrics)
+    extends EntityRepository[TaskFailure] {
 
-  def store(id: PathId, value: TaskFailure): Unit =
-    synchronized { taskFailures(id) = value }
+  def store(id: PathId, value: TaskFailure): Future[TaskFailure] =
+    super.storeWithVersion(id.safePath, value.version, value)
```

Comment:
We only expose one task failure in our API, so in theory we could just store that.

All our stores are "versioned" so I just implemented it like this. Our current API of the `EntityRepository` only supports "versioned" stores.

But you are right, these are not really versions. 

I don't feel that we should persist the lastTaskFailure for every app version we retain. This would be quite some effort (synchronizing this with the versions of the AppRepo etc) but wouldn't have such great effect. The lastTaskFailure itself contains information about the version of the task.

You are asking a lot of (valuable) questions without providing your opinion. What do you think is best?

These are the options I see:

A) Leave it as is. There is a technically unnecessary overhead for versioning, even though we do not use the different versions.
B) Extend the EntityRepository to support unversioned storage and use that.
C) Use the an `EntityStore` directly.

I think that A) might lead to performance problems in load situations but otherwise OK.

B) is probably needed for a better task storage anyway (we do not want to version our task infos, do we?)

C) is ugly.